File: problems/09_heap_priority_queue/0703_kth_largest_element_in_a_stream/solution.dart
// Problem: 0703. Kth Largest Element in a Stream
// Difficulty: Easy

class _MinHeap {
  final List<int> a = [];
  int get size => a.length;
  int get top => a[0];
  void push(int x) {
    // TODO: implement
  }
  int pop() {
    // TODO: implement
    return 0;
  }
}

class KthLargest {
  final int k;
  final _MinHeap h = _MinHeap();
  KthLargest(this.k, List<int> nums) {
    // TODO: implement
  }
  int add(int val) {
    // TODO: implement
    return 0;
  }
}


void main() {
  // Test 1
  // final result = solve(...);
  print('All tests passed!');
}

File: problems/09_heap_priority_queue/0703_kth_largest_element_in_a_stream/solution.py
"""
Problem: 0703. Kth Largest Element in a Stream
Difficulty: Easy
URL: https://leetcode.com/problems/kth-largest-element-in-a-stream/

解法：大小 k 的 min-heap，O(log k)/O(k)。
"""

from __future__ import annotations

from typing import List

import heapq

class KthLargest:
    def __init__(self, k: int, nums: List[int]):
        pass

    def add(self, val: int) -> int:
        pass


if __name__ == "__main__":
    kthLargest = KthLargest()

    # Test 1
    # result = kthLargest.solve(...)
    # assert result == expected

    print("All tests passed!")
